test/Serialization/Recovery/module-recovery-remarks.swift

```diff
@@ -15,6 +15,45 @@
 
 /// Main error downgraded to a remark.
 // CHECK-MOVED: LibWithXRef.swiftmodule:1:1: remark: reference to type 'MyType' broken by a context change; 'MyType' was expected to be in 'A', but now a candidate is found only in 'A_related'
+
+/// Contextual notes about the modules involved.
+// CHECK-MOVED: A.swiftmodule:1:1: note: declaration was expected to be found in this module
+// CHECK-MOVED: <binary format>
+// CHECK-MOVED: ^
+// CHECK-MOVED: module.modulemap:1:8: note: or found in this underlying module
+// CHECK-MOVED: module A {
+// CHECK-MOVED:        ^
+// CHECK-MOVED: A_related.swiftmodule:1:1: note: declaration was found in this module
+// CHECK-MOVED: <binary format>
+// CHECK-MOVED: ^
+
+/// More notes depending on the context
+// CHECK-MOVED: LibWithXRef.swiftmodule:1:1: note: this module was built with a Swift language version set to 5
+// CHECK-MOVED: while the current invocation uses 4
+
+// CHECK-MOVED: LibWithXRef.swiftmodule:1:1: note: the module 'LibWithXRef' enabled library-evolution; this file may need to be deleted if the SDK was modified
```

Comment:
Are these all notes of the previous remark?

Reply:
Yes they are. This is the worst case scenario, there won't be that many notes in most cases.